core/sr-arithmetic/src/parts_per_x.rs

```diff
@@ -0,0 +1,551 @@
+
+#[cfg(feature = "std")]
+use serde::{Serialize, Deserialize};
+
+use rstd::{ops, prelude::*, convert::TryInto};
+use codec::{Encode, Decode};
+use crate::traits::{SaturatedConversion, UniqueSaturatedInto, Saturating};
+
+/// Checks that `$x` is equal to `$y` with an error rate of `$error`.
+///
+/// # Example
+///
+/// ```rust
+/// # fn main() {
+/// sr_arithmetic::assert_eq_error_rate!(10, 10, 0);
+/// sr_arithmetic::assert_eq_error_rate!(10, 11, 1);
+/// sr_arithmetic::assert_eq_error_rate!(12, 10, 2);
+/// # }
+/// ```
+///
+/// ```rust,should_panic
+/// # fn main() {
+/// sr_arithmetic::assert_eq_error_rate!(12, 10, 1);
+/// # }
+/// ```
+#[macro_export]
+#[cfg(feature = "std")]
+macro_rules! assert_eq_error_rate {
```

Comment:
How did this end up here?

Reply:
yeah this is used in substrate as well, if you need it here and you want to stay independent, sadly you have to either duplicate it or come up with another way

Reply:
Would it make sense to move `assert_eq_error_rate` into `rstd`?

Reply:
I don't think so, rstd should be the representation of rust's `std`. 

It is used only in testing, Maybe you can put it in some testing util folder. 

But as I said, not a big concern as far as this PR is concerned.